_develop/connect/use-with-clarity.md

```diff
@@ -33,34 +26,78 @@ Connect allows you to open the authenticator with parameters indicating the deta
 </pre>
 </div>
 
-### `ContractCallOptions`
+* TOC
+{:toc}
+
+## How it works
+
+For your app's users to be able to execute a smart contract function, they need to sign and broadcast a transaction. It's important that users remain control of the private keys that sign these transactions. Connect provides an easy-to-use workflow that allows your users to securely sign transactions.
+
+Connect allows you to open the authenticator with parameters indicating the details of the transaction - like the smart contract address, function name, and specific arguments. Your users get the chance to see these details, and then sign and broadcast the transaction in a single click. Their transaction will be securely signed and broadcasted onto the Stacks blockchain. After this is done, a callback is fired to allow you to update your app.
+
+## Calling Clarity Contract Functions
+
+Once you have a Clarity smart contract built and deployed, you'll naturally want to allow your app's users to interact with it.
+
+To initiate a contract call transaction, use the `openContractCall` function.
+
+```ts
+import { openContractCall } from '@blockstack/connect';
+
+// While in beta, you must provide this option:
+const authOrigin = 'https://deploy-preview-301--stacks-authenticator.netlify.app';
+
+// Here's an example of options:
+const myStatus = 'hey there';
+const options = {
+  contractAddress: 'ST22T6ZS7HVWEMZHHFK77H4GTNDTWNPQAX8WZAKHJ',
+  contractName: 'status',
+  functionName: 'write-status!',
+  functionArgs: [
+    {
+      type: 'buff',
+      value: myStatus,
+    }
+  ],
+  authOrigin,
+  appDetails: {
+    name: 'SuperApp',
+    icon: 'https://example.com/icon.png'
+  },
+  finished: (data) => {
+    console.log('TX ID:', data.txId);
+    console.log('Raw TX:', data.txRaw);
+  },
+};
+
+await openContractCall(opts);
+```
 
-When signing a transaction, you need to specify a few details. Here is the exact interface that describes what options you have:
+When calling `openContractCall`, you need to specify a few details. Here is the exact interface that describes what options you have:
 
 ```ts
-export interface ContractCallOptions {
+interface ContractCallOptions {
   contractAddress: string;
   functionName: string;
   contractName: string;
   functionArgs?: any[];
   authOrigin?: string;
-  userSession?: UserSession;
-  appDetails?: AuthOptions['appDetails'];
-  finished?: (data: FinishedTxData) => void;
+  appDetails: AuthOptions['appDetails'];
```

Comment:
is this in reference to some class? Should link to it because I wouldn't know what the format here should be without the context

Reply:
seems like it's just name & icon. I'd explicitly put these in here

Reply:
This is the typescript interface for the appDetails key. It's using `AuthOptions['appDetails'];` so there is less duplication. We should link to the place that the `AuthOptions` interface is defined.

Reply:
I ended up just putting the explicit object interface as Alex suggested - it may be less technically exact, but it likely reduces confusion.